src/colorizer/TimeControls.ts

```diff
@@ -72,28 +88,22 @@ export default class TimeControls {
   }
 
   public handlePauseButtonClick(): void {
-    clearInterval(this.timerId);
-    this.timerId = DEFAULT_TIMER_ID;
+    if (this.timerId !== DEFAULT_TIMER_ID) {
+      clearTimeout(this.timerId);
+      this.timerId = DEFAULT_TIMER_ID;
+    }
+    this.currentlyPlaying = false;
     this.pauseCallbacks.forEach((callback) => callback());
   }
 
   public async handleFrameAdvance(delta: number = 1): Promise<void> {
     if (this.setFrameFn) {
-      this.setFrameFn(this.wrapFrame(this.canvas.getCurrentFrame() + delta));
-    }
-  }
-
-  public setIsDisabled(disabled: boolean): void {
-    this.isDisabled = disabled;
-    // Disable and clean up playback interval timer
-    if (disabled) {
-      clearInterval(this.timerId);
-      this.timerId = DEFAULT_TIMER_ID;
+      await this.setFrameFn(this.wrapFrame(this.canvas.getCurrentFrame() + delta));
     }
   }
 
   public isPlaying(): boolean {
-    return this.timerId !== -1;
+    return this.currentlyPlaying;
```

Comment:
is currentlyPlaying always in sync with (timerId !== -1) ?

Reply:
It wasn't guaranteed before, but I made some changes and removed `currentlyPlaying`.

Reply:
looks good